appsmithorg/appsmith
# fix: added loading state for chart widget

## Description

**Summary of changes**

- Added class `bp3-skeleton` when the `isLoading` prop is true.

**Motivation**

- A loading state did not existed when `Animate Loading` was set to true. This caused the widget to crash whenever the widget was attached with the query data. 

Fixes #10183

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

This fix is tested manually. Please find below the steps to test the fix:

- Add a chart widget on to the canvas.
- Attach a query to the chart's `Series Data` attribute for example, `{{Query1.data.map( (obj) =>{ return  {'x': obj.gender, 'y': obj.id } })}}`
- Set the property `Animate Loading` to true.
- Refresh the page.
- Observe that you will see a loader.

Please find below video for reference:

[![LOOM DEMO](http://cdn.loom.com/sessions/thumbnails/b9bf02ee25b34e47ba1460a8decc8462-00001.gif)](https://www.loom.com/share/b9bf02ee25b34e47ba1460a8decc8462)

## Checklist:






## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>


    // Code coverage diff between base branch:release and head branch: fix/fix-chart_widget_no_loading_state 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/widgets/ChartWidget/component/index.tsx | 49.41 **(0)** | 22.68 **(0.57)** | 50 **(0)** | 47.47 **(0)**</details>
